src/NServiceBus.Persistence.AzureStorage.ComponentTests/Sagas/When_executing_concurrently.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace NServiceBus.Persistence.AzureStorage.ComponentTests.Sagas
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.WindowsAzure.Storage;
+    using Microsoft.WindowsAzure.Storage.Table;
+    using NServiceBus.Sagas;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// These tests try to mimic different concurrent scenarios using two persiters trying to access the same saga.
+    /// </summary>
+    public class When_executing_concurrently
+    {
+        public When_executing_concurrently()
+        {
+            connectionString = AzurePersistenceTests.GetConnectionString();
+            var account = CloudStorageAccount.Parse(connectionString);
+            var client = account.CreateCloudTableClient();
+            cloudTable = client.GetTableReference(typeof(ConcurrentSagaData).Name);
+        }
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            await cloudTable.CreateIfNotExistsAsync().ConfigureAwait(false);
+            persister1 = new AzureSagaPersister(connectionString, true);
+            persister2 = new AzureSagaPersister(connectionString, true);
+
+            // clear whole table
+            var entities = await cloudTable.ExecuteQueryAsync(new TableQuery<TableEntity>()).ConfigureAwait(false);
+            foreach (var te in entities)
+            {
+                await cloudTable.DeleteIgnoringNotFound(te).ConfigureAwait(false);
+            }
+        }
+
+        [Test(Description = "Failed removal of the secondary index")]
+        public async Task Should_enable_inserting_when_only_secondary_exists()
```

Comment:
During the review with @Scooletz we thought it would make sense to split the test into two: 

1. To verify that a second persister doesn't blow up when is requesting a "primed" saga record when the actual saga is removed from the storage
1. To verify that when a saga is removed, another persister that had 2nd index for that saga cached, can update saga data for the same correlation property.